test/OpenTelemetry.Exporter.Geneva.Tests/LogSerializationTests.cs

```diff
@@ -0,0 +1,149 @@
+        var exportedFields = GetExportedFieldsAfterLogging(logger => logger.Log<object>(
+            logLevel: LogLevel.Information,
+            eventId: default,
+            state: null,
+            exception: new Exception(exceptionMessage),
+            formatter: null));
+
+        PrintFields(this.output, exportedFields);
+        var actualExceptionMessage = exportedFields["env_ex_msg"];
+        Assert.Equal(exceptionMessage, actualExceptionMessage);
+    }
+
+    private static void PrintFields(ITestOutputHelper output, Dictionary<object, object> fields)
+    {
+        foreach (var field in fields)
+        {
+            output.WriteLine($"{field.Key}:{field.Value}");
+        }
+    }
+
+    private static Dictionary<object, object> GetExportedFieldsAfterLogging(Action<ILogger> doLog)
+    {
+        Socket server = null;
+        string path = string.Empty;
+        try
+        {
+            var logRecordList = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                    .AddOpenTelemetry(options =>
+                    {
+                        options.AddInMemoryExporter(logRecordList);
+                    })
+                    .AddFilter(typeof(GenevaLogExporterTests).FullName, LogLevel.Trace)); // Enable all LogLevels
+
+            var logger = loggerFactory.CreateLogger<GenevaLogExporterTests>();
+            doLog(logger);
+
+            Assert.Single(logRecordList);
+            var exporterOptions = new GenevaExporterOptions();
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                exporterOptions.ConnectionString = "EtwSession=OpenTelemetry";
+            }
+            else
+            {
+                path = GenerateTempFilePath();
+                exporterOptions.ConnectionString = "Endpoint=unix:" + path;
+                var endpoint = new UnixDomainSocketEndPoint(path);
+                server = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.IP);
+                server.Bind(endpoint);
+                server.Listen(1);
+            }
+
+            using var exporter = new GenevaLogExporter(exporterOptions);
+            var m_buffer = typeof(GenevaLogExporter).GetField("m_buffer", BindingFlags.NonPublic | BindingFlags.Static).GetValue(exporter) as ThreadLocal<byte[]>;
+            _ = exporter.SerializeLogRecord(logRecordList[0]);
+            object fluentdData = MessagePack.MessagePackSerializer.Deserialize<object>(m_buffer.Value, MessagePack.Resolvers.ContractlessStandardResolver.Instance);
+
+            return GetFields(fluentdData);
+        }
+        finally
+        {
+            server?.Dispose();
+            try
+            {
+                File.Delete(path);
+            }
+            catch
+            {
+            }
+        }
+    }
+
+    private static string GenerateTempFilePath()
+    {
+        while (true)
```

Comment:
Why do we need a loop here? (what's the semantic of this method?)

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/test/OpenTelemetry.Exporter.Geneva.Tests/GenevaLogExporterTests.cs#L1054

Copied existing as is. As part of refactoring, will revisit if this (or any other such helpers) needs changes.